src/components/AccountSelector/AccountSelector.tsx

```diff
@@ -0,0 +1,176 @@
+import { Amount } from '@/components/Amount/Amount'
+import { AssetIcon } from '@/components/AssetIcon'
+import { MiddleEllipsis } from '@/components/MiddleEllipsis/MiddleEllipsis'
+import { useLocaleFormatter } from '@/hooks/useLocaleFormatter/useLocaleFormatter'
+import { bnOrZero } from '@/lib/bignumber/bignumber'
+import { fromBaseUnit } from '@/lib/math'
+import { isUtxoAccountId } from '@/lib/utils/utxo'
+import { accountIdToLabel } from '@/state/slices/portfolioSlice/utils'
+import {
+  selectAssetById,
+  selectMarketDataByAssetIdUserCurrency,
+  selectPortfolioAccountBalancesBaseUnit,
+  selectPortfolioAccountIdsByAssetIdFilter,
+} from '@/state/slices/selectors'
+import { useAppSelector } from '@/state/store'
+
+export type AccountSelectorProps = {
+  assetId: AssetId
+  accountId?: AccountId
+  onChange: (accountId: AccountId) => void
+  disabled?: boolean
+  autoSelectHighestBalance?: boolean
+  buttonProps?: ButtonProps
+  boxProps?: BoxProps
+}
+
+const chevronIconSx = {
+  svg: { h: '18px', w: '18px' },
+}
+
+export const AccountSelector: FC<AccountSelectorProps> = memo(
+  ({
+    assetId,
+    accountId: selectedAccountId,
+    onChange,
+    disabled,
+    autoSelectHighestBalance,
+    buttonProps,
+    boxProps,
+  }) => {
+    const { isOpen, onOpen, onClose } = useDisclosure()
+    const {
+      number: { localeParts },
+    } = useLocaleFormatter()
+
+    const filter = useMemo(() => ({ assetId }), [assetId])
+    const accountIds = useAppSelector(state =>
+      selectPortfolioAccountIdsByAssetIdFilter(state, filter),
+    )
+    const asset = useAppSelector(state => selectAssetById(state, assetId))
+    const accountBalances = useAppSelector(selectPortfolioAccountBalancesBaseUnit)
+    const marketData = useAppSelector(state =>
+      selectMarketDataByAssetIdUserCurrency(state, assetId),
+    )
+
+    const marketDataPrice = useMemo(() => marketData?.price ?? 0, [marketData])
+
+    useEffect(() => {
+      if (!selectedAccountId) {
+        onChange(accountIds[0])
+      }
+    }, [selectedAccountId, accountIds, onChange])
+
+    const selectedAccountDetails = useMemo(() => {
+      if (!selectedAccountId || !asset) return null
+
+      const cryptoBalance = fromBaseUnit(
+        accountBalances?.[selectedAccountId]?.[assetId] ?? 0,
+        asset.precision ?? 0,
+      )
+      const fiatBalance = bnOrZero(cryptoBalance).times(marketDataPrice)
+
+      return {
+        cryptoBalance,
+        fiatBalance,
+        label: isUtxoAccountId(selectedAccountId) ? accountIdToLabel(selectedAccountId) : undefined,
+      }
+    }, [selectedAccountId, asset, accountBalances, assetId, marketDataPrice])
+
+    const handleAccountSelect = useCallback(
+      (accountId: AccountId) => {
+        onChange(accountId)
+        onClose()
+      },
+      [onChange, onClose],
+    )
+
+    const isButtonDisabled = disabled || accountIds.length <= 1
```

Comment:
stylistic: preference for `accountIds.length === 1`

Not a difference at runtime since we already guard 0-length with `if (!accountIds.length) return null` below, but keep things declarative re: what's *actually* the intent here.